Cleanup/remove legacy assets post bs5 migration

labmecanicatec opened the pull request:
This PR includes a major cleanup and modernization of the codebase focused on Bootstrap 5 migration, UI simplification, and compatibility improvements. Key changes include:

- Removal of unused CSS, LESS, images, and JS libraries.
- Replacement of legacy assets (e.g., Font Awesome, ColorPicker) with Bootstrap 5 alternatives.
- Refactoring of templates and layout for better readability and maintainability.
- Implementation of DataTables in place of legacy pagination methods.
- Fixes to PayPal payment logic for API compatibility and deprecated behavior (e.g., NumberFormatter).
- Enabled searchClear in filters for better UX.
- Fixed modal dialog issues caused by conflicting scripts.

These updates improve performance, reduce maintenance overhead, and align the application with modern frontend practices.

JohnVillalovos commented on Web/scripts/js/jquery.contextMenu.js:
```diff
@@ -1,1686 +1,2122 @@
+ * jQuery contextMenu v2.7.0 - Plugin for simple contextMenu handling
  *
- * Version: git-master
+ * Version: v2.7.0
```
Why isn't this the v2.9.2 version like `jquery.contextMenu.css`?

JohnVillalovos replied:
I have updated this to be the 2.9.2 version.

JohnVillalovos commented on tpl/Admin/Groups/manage_groups.tpl:
@labmecanicatec It would be easier to review this if it didn't do formatting changes at the same time.

JohnVillalovos commented on tpl/Admin/Reservations/manage_reservations.tpl:
@labmecanicatec It would be easier to review this if it didn't do formatting changes at the same time.

JohnVillalovos commented on tpl/register.tpl:
@labmecanicatec It would be easier to review this if it didn't do formatting changes at the same time.

JohnVillalovos reviewed (changes requested):
Thanks @labmecanicatec Nice cleanup.

I left some feedback.

labmecanicatec committed:
refactor: remove legacy pagination functions

- Removed obsolete functions CreatePageLink and CreatePagination as
  pagination is now handled via DataTables.
- Fixed deprecated usage of NumberFormatter::formatCurrency() by
  ensuring a valid float is always passed.
- Improved compatibility with PHP 8.2+ and reduced runtime warnings.

The application now relies entirely on client-side pagination and rendering through DataTables.

JohnVillalovos reviewed (approved):
LGTM

Thanks @labmecanicatec
